src/Template/TemplateSection.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace League\Plates\Template;
+
+/**
+ * Defines the object format of the data target 
+ * to be rendered in a Template's Section.
+ */
+class TemplateSection
+{
+    /**
+     * Content history.
+     *
+     * @var array
+     */
+    protected $mutationlog = [];
+
+    /**
+     * Section name.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @param string  $name
+     * @param ?string $content
+     * @param ?int    $mode
+     */
+    public function __construct(string $name, $content = '', $mode = Template::SECTION_MODE_REWRITE)
+    {
+        $this->name = $name;
+        if (!empty($content)) {
+            $this->add($content, $mode);
+        }
+    }
+
+    /**
+     * @param  string $content
+     * @param  int    $mode
+     * @return void
+     */
+    public function add(string $content, int $mode)
+    {
+        $this->mutationlog[] = new TemplateSectionMutation($mode, $content);
+    }
+
+    public function merge($other)
+    {
+        if ($other->name !== $this->name) {
+            throw new \InvalidArgumentException("Only same section may be merged");
+        }
+
+        $this->mutationlog = array_merge($this->mutationlog, $other->mutationlog);
```

Comment:
I'm still unsure on what to do if multiple rewrite exists, some in the right mutationlog. If we simply merge, the operation on `$this->mutationlog` will be lost even if APPEND and PREPEND exists.

Reply:
I'm thinking of grouping REWRITE before any mutation. That is consistent with what template system should do in my mind.

layout.php
```php
<?php $this->section('javascript') ?>
```

page.php
```php
<?php $this->layout('layout') ?>

<?php $this->start('javascript') ?>
  <script src="base-page-script-very-light-weight.js"></script>
<?php $this->end() ?>

<?= $this->fetch('common-component-a') ?>
<?= $this->fetch('heavyweight-component-b') ?>
```

heavyweight-component-b.php
```php
<?php $this->start('javascript') ?>
  <script src="base-page-script-support-heavy-weight.js"></script>
<?php $this->end() ?>
```

common-component-a.php
```php
<?php $this->push('javascript') ?>
  <script src="common-component-a.js"></script>
<?php $this->end() ?>
```

The end result will be,
```html
<script src="base-page-script-support-heavy-weight.js"></script>
<script src="common-component-a.js"></script>
```